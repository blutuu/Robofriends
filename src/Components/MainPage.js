import React, { Component } from "react";
import "./MainPage.css";
import CardList from "../Components/CardList";
import SearchBox from "../Components/SearchBox";
import Scroll from "../Components/Scroll";
import ErrorBoundary from "../Components/ErrorBoundary";
import CounterButton from "../Components/CounterButton";

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filteredRobots = () => {
    return this.props.robots.filter((robot) => {
      return robot.name
        .toLowerCase()
        .includes(this.props.searchField.toLowerCase());
    });
  };

  render() {
    const { onSearchChange, isPending } = this.props;

    return isPending ? (
      <h1>LOADING...</h1>
    ) : (
      <div className="App">
        <h1 className="f2">Robofriends</h1>
        <SearchBox searchChange={onSearchChange} />
        <CounterButton />
        <Scroll>
          <ErrorBoundary>
            <CardList robots={this.filteredRobots()} />
          </ErrorBoundary>
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
