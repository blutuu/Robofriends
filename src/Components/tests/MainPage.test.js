import React from "react";
import { shallow } from "enzyme";
import MainPage from "../../Components/MainPage";

let wrapper;

beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    searchField: "",
    onSearchChange: jest.fn(),
    robots: [],
    isPending: false,
  };

  wrapper = shallow(<MainPage {...mockProps} />);
});

describe("Testing main page component", () => {
  it("renders MainPage without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("filters robots correctly", () => {
    const mockProps2 = {
      onRequestRobots: jest.fn(),
      searchField: "a",
      onSearchChange: jest.fn(),
      robots: [
        {
          id: 3,
          name: "John",
          email: "john@gmail.com",
        },
      ],
      isPending: false,
    };

    const filteredRobots = [];
    const wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(wrapper2.instance().filteredRobots()).toEqual(filteredRobots);
  });

  it("filters robots correctly 2", () => {
    const mockProps3 = {
      onRequestRobots: jest.fn(),
      searchField: "j",
      onSearchChange: jest.fn(),
      robots: [
        {
          id: 3,
          name: "John",
          email: "john@gmail.com",
        },
      ],
      isPending: false,
    };

    const filteredRobots2 = [
      {
        id: 3,
        name: "John",
        email: "john@gmail.com",
      },
    ];
    const wrapper3 = shallow(<MainPage {...mockProps3} />);
    expect(wrapper3.instance().filteredRobots()).toEqual(filteredRobots2);
  });

  it("tests the pending variable", () => {
    const mockProps4 = {
      onRequestRobots: jest.fn(),
      searchField: "",
      onSearchChange: jest.fn(),
      robots: [],
      isPending: true,
    };

    const wrapper4 = shallow(<MainPage {...mockProps4} />);
    expect(wrapper4).toMatchSnapshot();
  });
});
