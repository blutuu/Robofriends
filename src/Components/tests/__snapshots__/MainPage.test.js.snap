// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing main page component renders MainPage without crashing 1`] = `
<div
  className="App"
>
  <h1
    className="f2"
  >
    Robofriends
  </h1>
  <SearchBox
    searchChange={[MockFunction]}
  />
  <CounterButton />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;

exports[`Testing main page component tests the pending variable 1`] = `
<h1>
  LOADING...
</h1>
`;
